commands/ssh.go

```diff
@@ -156,7 +160,32 @@ func RunSSH(c *CmdConfig) error {
 		return errors.New("Could not find Droplet address")
 	}
 
+	retry, err := c.Doit.GetBool(c.NS, doctl.ArgCommandRetry)
+	if err != nil {
+		return err
+	}
+
 	runner := c.Doit.SSH(user, ip, keyPath, port, opts)
```

Comment:
It might be interesting to push the retry logic into the SSH runner itself. You can pass configuration options on the command line to SSH with the `-o` flag, e.g. `-o ConnectionAttempts=5`

> **ConnectionAttempts**
> Specifies the number of tries (one per second) to make before exiting. The argument must be an integer. This may be  useful in scripts if the connection sometimes fails. The default is 1.

https://linux.die.net/man/5/ssh_config

SSH itself probably has a a better understanding of what error conditions are retryable.

Reply:
Oh great catch. Pushing it down to the SSH runner is working out well actually. Thanks for the suggestion!